src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -384,6 +384,41 @@ public ComplaintResource(AuthorizationCheckService authCheckService, ExerciseRep
         return ResponseEntity.ok(getComplaintsByComplaintType(complaints, complaintType));
     }
 
+    /**
+     * Get /courses/:courseId/exams/:examId/complaints
+     * <p>
+     * Get all the complaints filtered by courseId, complaintType and optionally tutorId.
+     * @param examId the id of the tutor by which we want to filter
+     * @param courseId the id of the course we are interested in
+     * @param complaintType the type of complaints we are interested in
+     * @return the ResponseEntity with status 200 (OK) and a list of complaints. The list can be empty
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/complaints")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<Complaint>> getComplaintsByCourseIdAndExamId(@PathVariable Long courseId, @PathVariable Long examId, @RequestParam ComplaintType complaintType) {
+        // Filtering by courseId
+        Course course = courseRepository.findByIdElseThrow(courseId);
+
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        boolean isAtLeastTutor = authCheckService.isAtLeastTeachingAssistantInCourse(course, user);
+        boolean isAtLeastInstructor = authCheckService.isAtLeastInstructorInCourse(course, user);
+
+        if (!isAtLeastTutor) {
+            throw new AccessForbiddenException("Insufficient permission for these complaints");
+        }
+        if (!isAtLeastInstructor) {
+            // At the moment the complete list of all exam-complaints should only be visible for instructors
+            throw new AccessForbiddenException("Insufficient permission for these complaints");
+        }
+
+        List<Complaint> complaints;
+
+        complaints = complaintService.getAllComplaintsByExamId(examId);
```

Comment:
move to the previous line: List<Complaint> complaints = complaintService.getAllComplaintsByExamId(examId);

Reply:
Did that in the style improvement commit :+1: